### PR TITLE
Add krnl prefetch

### PR DESCRIPTION
**Pull request by AlexandreEichenberger**
Cannot add `affine.prefetch` during ONNX to KRNL lowering because affine prefetch check the enclosing loop nests, which are not yet `affine.for` loops during KRNL lowering. 

Thus created a placeholder `krnl.prefetch` that will later be lowered to `affine.prefetch`.

### CODE REVIEW

**Comment by AlexandreEichenberger** [comment]
cleaned up old dialect builder for affine prefetch that are not in use.

**Review by tungld — APPROVED** [review]
LGTM.

File: src/Dialect/Krnl/KrnlOps.cpp [hunk 1]
```diff
@@ -1029,6 +1029,111 @@ void KrnlGetLinearOffsetIndexOp::print(OpAsmPrinter &p) {
...
+    Value memref, bool isWrite, unsigned localityHint, bool isDataCache) {
+  build(builder, result, memref, {}, isWrite, localityHint, isDataCache);
+}
+
+//
+// krnl.prefetch %0[%i, %j + 5], read, locality<3>, data : memref<400x400xi32>
+// Code lifted from affine prefetch as is.
+// I have seen parsing errors when multiple '#x' are used in the indices, could
+// not tell why.
+//   krnl.prefetch %arg0[%1#0, %1#1, %3], read, locality<3>, data :
```
Comment (chentong319) [inline src/Dialect/Krnl/KrnlOps.cpp:1073]: How about `affine.prefetch %arg0[%1#0, %1#1, %3], read, locality<3>, data ` ?

**Comment by AlexandreEichenberger** [comment]
Got included as part of other PRs

**Pull request closed without merging**